ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -123,189 +116,189 @@ public SECDataConverter(string rawSource, string destination, string knownEquity
-                            {
-                                return;
-                            }
+                            newTextLine = $"{newTextLine.TrimEnd()} 00:00:00";
+                        }
 
-                            WriteReport(reports, ticker);
+                        // Encode all contents inside tags to prevent errors in XML parsing.
+                        // The json deserializer will convert these values back to their original form
+                        if (!parsingText && HasValue(newTextLine))
+                        {
+                            newTextLine =
+                                $"<{currentTagName}>{SecurityElement.Escape(GetTagValueFromLine(newTextLine))}</{currentTagName}>";
+                        }
+                        // Escape all contents inside TEXT tags
+                        else if (parsingText)
+                        {
+                            newTextLine = SecurityElement.Escape(newTextLine);
                         }
-                    );
 
-                    // This will clean up after ourselves without having to pay
-                    // the expense of deleting every single file inside the raw_data folder
-                    Directory.Delete(extractDataPath, true);
+                        // Don't encode the opening tag
+                        if (currentTagName == "TEXT")
+                        {
+                            parsingText = true;
+                        }
+
+                        xmlText.AppendLine(newTextLine);
+                    }
+
+                    ISECReport report;
+                    try
+                    {
+                        report = factory.CreateSECReport(xmlText.ToString());
+                    }
+                    // Ignore unsupported form types for now
+                    catch (DataException)
+                    {
+                        return;
+                    }
+                    catch (XmlException e)
+                    {
+                        Log.Error(e, $"SECDataConverter.Process(): Failed to parse XML from file path: {rawReportFilePath}");
+                        return;
+                    }
+                    catch (Exception e)
+                    {
+                        Log.Error(e, "SECDataConverter.Process(): Unknown error encountered");
+                        return;
+                    }
+
+                    // First filer listed in SEC report is usually the company listed on stock exchanges
+                    var companyCik = report.Report.Filers.First().CompanyData.Cik;
+
+                    // Some companies can operate under two tickers, but have the same CIK.
+                    // Don't bother continuing if we don't find any tickers for the given CIK
+                    List<string> tickers;
+                    if (!CikTicker.TryGetValue(companyCik, out tickers))
+                    {
+                        return;
```

Comment:
Again, logging here may be nice, especially if this program will eventually be run as part of a daily update process -- it's nice to catch major decisions (such as not processing an item) in the logs so that in the event something unexpected happens elsewhere in the stack, we're able to figure out what this program did with that data.

Also, if the reason for not logging is that it's _too much_ -- we can hold these kinds of things in RAM and at the end write a summary to the log of what the result of processing each item was (or to some other summary specific files, though my preference would be the standard log file)

Reply:
The raw amount of CIKs included with the SEC reports is in my opinion way too much to even contain in RAM. I'd say that for a good chunk of each day, there are companies that aren't listed on exchanges that submit filings with the SEC, and are useless in our use case.

On another note, we're operating at very high RAM capacity, and we don't want to incur any more memory usage. If we had infinite resources, I'd for sure consider logging every event. But given that this PR's branch name has `reduce-log-spam-secdataconverter`, I guess you can see where I was going with this 😛